evalml/tests/automl_tests/test_automl.py

```diff
@@ -415,17 +416,19 @@ def test_automl_bad_data_check_parameter_type():
         automl.search(data_checks=[MockDataCheckErrorAndWarning])
 
 
-@patch('evalml.pipelines.RegressionPipeline.fit')
```

Comment:
Hmmmm. I suspect that this isn't actually testing what we want it to. We mock `InvalidTargetDataCheck` to return "R2", and then test that the default objective name is "R2", but the two aren't necessarily related. Let's delete this for now, and I can file an issue to track testing this if I'm understanding this correctly.